modules/signals/entities/src/with-entities.ts

```diff
@@ -0,0 +1,67 @@
+import { computed } from '@angular/core';
+import {
+  SignalStoreFeature,
+  signalStoreFeature,
+  withComputed,
+  withState,
+} from '@ngrx/signals';
+import {
+  EntityId,
+  EntityMap,
+  EntitySignals,
+  EntityState,
+  NamedEntitySignals,
+  NamedEntityState,
+} from './models';
+import { getEntityStateKeys } from './helpers';
+
+export function withEntities<Entity>(): SignalStoreFeature<
+  { state: {}; signals: {}; methods: {} },
+  {
+    state: EntityState<Entity>;
+    signals: EntitySignals<Entity>;
+    methods: {};
+  }
+>;
+export function withEntities<Entity, Collection extends string>(config: {
+  entity: Entity;
+  collection: Collection;
+}): SignalStoreFeature<
+  { state: {}; signals: {}; methods: {} },
+  {
+    state: NamedEntityState<Entity, Collection>;
+    signals: NamedEntitySignals<Entity, Collection>;
+    methods: {};
+  }
+>;
+export function withEntities<Entity>(config: {
+  entity: Entity;
+}): SignalStoreFeature<
+  { state: {}; signals: {}; methods: {} },
+  {
+    state: EntityState<Entity>;
+    signals: EntitySignals<Entity>;
+    methods: {};
+  }
+>;
+export function withEntities<Entity>(config?: {
```

Comment:
Yes, `withEntities<EntityType>()` is preferred when a store has only one collection. For multiple (named) collections, we can use config:

```ts
const BookStore = signalStore(
  withEntities({ entity: type<Book>(), collection: 'book' }),
  withEntities({ entity: type<Author>(), collection: 'author' })
);
```

> **Note:** Partial type inference is still not supported in TypeScript, so it's not possible to e.g. explicitly pass entity type and infer collection string literal from the input argument. In other words, this will not work:
> ```ts
> // error: we need to explicitly pass both generics or infer both of them from input
> withEntities<User>({ collection: 'user' }) 
>```

Btw, I found defining config as a separate object useful, so it can be reused for updaters:

```ts
const booksMeta = {
  entity: type<Book>(),
  collection: 'book',
  idKey: 'isbn', // custom id key
} as const;

const authorsMeta = {
  entity: type<Author>(),
  collection: 'author',
} as const;

const BookStore = signalStore(
  withEntities(booksMeta),
  withEntities(authorsMeta)
);

const booksStore = inject(BooksStore);

patchState(
  booksStore,
  setAllEntities(books, booksMeta),
  setAllEntities(authors, authorsMeta)
);
```

Regarding type check, it's possible to display a compilation error if the `Entity` generic is `unknown` (not specified), but I didn't want to introduce additional complexity with types. Currently, if not specified, entity type will be inferred as `unknown` (which is fine IMO):

```ts
const Store = signalStore(withEntities());

const store = inject(Store);

store.entityMap() // type: EntityMap<unknown>
store.entities() // type: unknown[]
```

Reply:
> Btw, I found defining config as a separate object useful, so it can be reused for updaters:

That's also what I was thinking from seeing the tests.

> Regarding type check, it's possible to display a compilation error if the Entity generic is unknown (not specified), but I didn't want to introduce additional complexity with types. Currently, if not specified, entity type will be inferred as unknown (which is fine IMO):

👍 For keeping the types simple.